integration-test/background/click-to-load-youtube.js

```diff
@@ -244,6 +258,152 @@ describe('Test YouTube Click To Load', () => {
 
         await page.close()
     })
+
+    it('CTL: YouTube Preview', async () => {
+        const waitForExpectedElement = dataKey =>
+            page.waitForFunction((expectedDataKey) => (
+                Boolean(document.querySelector(`[data-key=${expectedDataKey}]`))
+            ),
+            { polling: 10 }, dataKey)
+
+        // Open the test page and start logging network requests.
+        const page = await browser.newPage()
+        const pageRequests = []
+        const clearRequests = await logPageRequests(page, pageRequests)
+
+        // Navigate to test page
+        await pageWait.forGoto(page, testSite)
+
+        // Once the user clicks to preview a video, should change from blocked state
+        // to preview state and request only YouTube Preview images
+        clearRequests()
+        {
+            const totalEmbeddedVideosBlocked = (await page.$$('div[id^=yt-ctl-dialog-')).length
+
+            // Click toggle to enable previews
+            const previewToggle = await page.evaluateHandle(
+                'document.querySelector("div:nth-child(2) > div")' +
+                '.shadowRoot.querySelector("button[data-key=yt-preview-toggle]")'
+            )
+            await previewToggle.click()
+            await waitForExpectedElement('modal')
+            // Click modal button to confirm enabling YT previews
+            const enablePreviewsButton = await page.evaluateHandle(
+                'document.querySelector("body > div[data-key=modal]")' +
+                    '.shadowRoot.querySelector("button[data-key=allow]")'
+            )
+            await enablePreviewsButton.click()
+            await pageWait.forNetworkIdle(page)
+
+            const totalEmbeddedVideosPreviews = (await page.$$('div[id^=yt-ctl-preview-')).length
+
+            expect(totalEmbeddedVideosBlocked).toEqual(totalEmbeddedVideosPreviews)
+        }
+
+        {
+            const {
+                youTubeIframeApi, youTubeStandard, youTubeNocookie, youTubeImage
+            } = summariseYouTubeRequests(pageRequests)
+
+            expect(youTubeIframeApi.checked).toBeFalse()
+            expect(youTubeIframeApi.alwaysRedirected).toBeTrue()
+            expect(youTubeStandard.blocked).toEqual(0)
+            expect(youTubeNocookie.blocked).toEqual(0)
+            expect(youTubeNocookie.allowed).toEqual(0)
+            expect(youTubeImage.blocked).toEqual(0)
+            expect(youTubeImage.allowed).toEqual(youTubeImage.total)
```

Comment:
Maybe check that `youtubeImage.total` is > 0? Or do we somehow already know that at least one YouTube image has loaded?